### PR TITLE
Added a 'CONTRIBUTING.md' file

### PR DESCRIPTION
**Pull request by dstorrs**

### CODE REVIEW

**Comment by mflatt** [comment]
I like the idea of adding a "CONTRIBUTING.md", but I wonder whether it should instead contain links to existing resources, especially
  https://blog.racket-lang.org/2017/09/tutorial-contributing-to-racket.html

Stepping back a little, this PR's first cut at "CONTRIBUTING.mb" doesn't seem to be Racket-specific. If there's a good site that already explains how to work with GitHub and PRs, then we could link to that. But the blog post linked above seems to cover the same material and more.

**Comment by dstorrs** [comment]
I have no investment in the content of this file.  If you want it to simply
say "If you want to contribute, read this post" then I'm fine with it.  You
should have edit access on the PR, so feel free to change it to whatever
you want; if you don't, let me know and I'll change it.

**Comment by default-kramer** [comment]
+1 for https://blog.racket-lang.org/2017/09/tutorial-contributing-to-racket.html

That is exactly what I wish I had seen when I made my first contribution recently.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+* Make your changes
+* Scroll to the bottom of the page, enter a message that describes what you did
+* Choose the radio button that says "*Create a new branch for this commit and start a pull request. Learn more about pull requests*"  It will propose a branch name for you.
+* Click the green "*Propose File Change*"
+* Click on the 'Pull Requests' button, just below the name of the repository
+* Find the branch name that you just created, click 'Compare and pull request'
+* Add some explanatory text to the message box, explaining what you did
+* Click 'Create Pull Request'
+* Bask in the glory of helping the Racket Community
+
```
Comment (euhmeuh) [inline CONTRIBUTING.md]:
```suggestion
# Where to go from there?

You can read the [Racket contribution guide](https://blog.racket-lang.org/2017/09/tutorial-contributing-to-racket.html).
```

Comment (euhmeuh) [inline CONTRIBUTING.md]:
```suggestion
# Where to go from there?

You can read the [Racket contribution guide](https://blog.racket-lang.org/2017/09/tutorial-contributing-to-racket.html).
```

**Comment by pmatos** [comment]
Can I suggest adding this to the wiki instead and closing the PR? @spdegabrielle has been working on the wiki as of late and could suggest a good place for this information.

**Comment by pmatos** [comment]
@spdegabrielle added some of the information on the suggested file to https://github.com/racket/racket/wiki/Contributing-to-Racket

I will try to proprose a new contribution file.

**Comment by spdegabrielle** [comment]
Not having any 'CONTRUBUTING' file risks turning away potential contributors. 

@mflatt while this could be improved, it is helpful, and isn't damaging. 
- the blog post is linked
- the information is accurate
- it has the advantage over the blog post of beign in the same location as the repository. 
- given the centrality of github oss it is likely that potential contributors will find the respostory *before* they find the blog post./. 

@pmatos While the wiki is useful, it is standard practice to include a 'CONTRIBUTING' file in a repo. 

I'd propose that this is merged immediately 'as-is' as it can only help potential contributors. 

'Perfect is the enemy of good'

Kind regards, 
Stephen

**Comment by spdegabrielle** [comment]
Addressing the second part of @mflatt  comment; 
> If there's a good site that already explains how to work with GitHub and PRs, then we could link to that.

https://help.github.com/en/articles/about-pull-requests
https://help.github.com/en/articles/creating-a-pull-request

I've updated https://github.com/racket/racket/wiki/Ways-to-contribute-to-Racket

**Comment by mflatt** [comment]
@bennn Thanks for moving this along, but another thread on this point has already moved along, and we now have some redundancy to clean up.

There was already a "CONTRIBUTING.md" before this push, but in ".github" instead of the root directory. (GitHub recognizes ".github/CONTRIBUTING.md" the same as a root "CONTRIBUTING.md".) The file was moved in ".github" as part of an effort to declutter the root directory and have a clearer entry point as just "README.md".

Meanwhile, "README.md" now has a section on contributing. Both "README.md" and ".github/CONTRIBUTING.md" point to "build.md", which is a markdown rendering of https://docs.racket-lang.org/racket-build-guide/index.html . In general, the idea was to move information on building and contributing into a fully fledged Racket document.

So, I think we should remove the root "CONTRIBUTING.md" again, but make sure that the right information is in "build.md". Although I did check this PR to help fill in the initial "Contributing" section of "build.md", I was hoping that others would take a look and refine it, and you may see that some information is still missing and needs to be added. The source of "build.md" is in "pkgs/racket-build-guide".

**Comment by bennn** [comment]
Oops --- I completely missed that thread. I'll delete the root CONTRIBUTING.md and look over build.ml

**Comment by mflatt** [comment]
Thanks! BTW, when you modify "pkgs/racket-build-guide", run `make` in that directory to update "build.md".

**Pull request merged**